lightning/src/chain/keysinterface.rs

```diff
@@ -869,9 +891,15 @@ impl KeysManager {
 				};
 				let channel_master_key = master_key.ckd_priv(&secp_ctx, ChildNumber::from_hardened_idx(3).unwrap()).expect("Your RNG is busted");
 				let rand_bytes_master_key = master_key.ckd_priv(&secp_ctx, ChildNumber::from_hardened_idx(4).unwrap()).expect("Your RNG is busted");
-				let inbound_payment_key: SecretKey = master_key.ckd_priv(&secp_ctx, ChildNumber::from_hardened_idx(5).unwrap()).expect("Your RNG is busted").private_key.key;
-				let mut inbound_pmt_key_bytes = [0; 32];
-				inbound_pmt_key_bytes.copy_from_slice(&inbound_payment_key[..]);
+				let (inbound_payment_key, phantom_secret) = if let Some(key) = cross_node_seed {
+					let (inbound_key, phantom_key) = hkdf_extract_expand_twice(b"LDK Inbound and Phantom Payment Key Expansion", key);
```

Comment:
It's not readily clear to me why the `inbound_payment_key` derivation needs to change for this. Aside: this is where commit messages explaining the _why_ in more detail are useful for both reviewers and posterity. 🙂

Reply:
Thanks, updated the commitment message so hopefully it's a bit clearer